private/buf/cmd/buf/buf.go

```diff
@@ -150,23 +148,36 @@ func NewRootCommand(name string) *appcmd.Command {
 				Use:   "mod",
 				Short: "Configure and update buf modules.",
 				SubCommands: []*appcmd.Command{
-					modinit.NewCommand("init", builder, "", false),
+					modinit.NewCommand("init", builder, modInitDeprecationMessage, true), // TODO(alex): Think about this one (see TODO below).
+					export.NewCommand("export", builder, "", false),                      // TODO(alex): Move this command into the modexport package.
 					modprune.NewCommand("prune", builder),
+					push.NewCommand("push", builder, "", false), // TODO(alex): Move this command into the modpush package.
 					modupdate.NewCommand("update", builder, "", false),
 					modclearcache.NewCommand("clear-cache", builder, "", false, "cc"),
 				},
 			},
+			// TODO(alex): We might actually want to keep "buf mod init" in the "mod" sub-command since it's directly tied to
+			// the other "mod" commands. For example, I initialize a module with "buf mod init", then push it immediately after
+			// with "buf mod push". Otherwise, we would need to jump between multiple sub-commands (initialize with "config",
+			// then push with "mod").
+			//
+			// However, keeping it in mod is awkward for "config ls-{breaking,lint}-rules" - listing the content of the `buf.yaml`
+			// should exist alongside how it's initialized.
+			//
+			// How do we foresee future configuration commands for `buf.work.yaml` and `buf.gen.yaml` to look? Will these be new
+			// sub-commands under the "config" sub-command? What does that suggest about how we should structure this (i.e. "buf config mod-init")?
 			{
 				Use:   "config",
 				Short: "Interact with the configuration of Buf.",
 				SubCommands: []*appcmd.Command{
+					modinit.NewCommand("init", builder, "", false), // TODO(alex): Move this command into the configinit package.
 					configlslintrules.NewCommand("ls-lint-rules", builder, "", false),
 					configlsbreakingrules.NewCommand("ls-breaking-rules", builder, "", false),
```

Comment:
We can keep the `ls-{breaking,lint}-rules` commands as-is, and move them into the `buf mod` sub-command later on if it makes more sense. In short, we're fine with the cost of deprecating these commands and hiding them later on, and this doesn't need to be part of the v1 release.